### PR TITLE
add non-exported helper `offset_to`

### PR DESCRIPTION
**Pull request by johnnychen94**
I recently need to align all arrays into a common origin so I figure it can be helpful to put this here. I'll add docs and tests if you think it's a good idea.

### CODE REVIEW

File: src/OffsetArrays.jl [hunk 1]
```diff
@@ -768,6 +768,26 @@ centered(A::AbstractArray, cp::Dims=center(A)) = OffsetArray(A, .-cp)
...
+
+Shifting the axes of all arrays in `As` so that their origin matches `X`. The output will be
+a tuple of `OffsetArray`s.
+
+```jldoctest; setup=:(using OffsetArrays)
+julia> X = OffsetArray(rand(4, 4), (2, 3)); # origin: (3, 4)
+
+julia> A1, A2, A3 = rand(4, 4), rand(2, 3), OffsetArray(rand(3, 2), (-1, -1)); # all have different axes
+
+julia> A1o, A2o, A3o = OffsetArrays.offset_to(X, A1, A2, A3);
```
Comment (johnnychen94) [inline src/OffsetArrays.jl:782]: Should this return `Xo, A1o, A2o, A3o` or just `A1o, A2o, A3o`?

**Comment by jishnub** [comment]
Is this really something that we need here? I feel a `map` is easier to read in this case.

```julia
julia> A1, A2, A3 = rand(4, 4), rand(2, 3), OffsetArray(rand(3, 2), (-1, -1)); # all have different axes

julia> A1o, A2o, A3o = map((A1, A2, A3)) do A
                           OffsetArray(A, OffsetArrays.Origin(1,2))
                       end;

julia> first.(axes(A1o))
(1, 2)

julia> first.(axes(A1o)) == first.(axes(A2o)) == first.(axes(A3o))
true
```

**Comment by johnnychen94** [comment]
I won't say this is necessary; it's just another small helper function like `Origin` and `center`/`centered` we build.

**Comment by jishnub** [comment]
Maybe we can change the behavior of `Origin`, such that 

```julia
julia> OffsetArrays.Origin(1,2)(A) == OffsetArray(A, OffsetArrays.Origin(1,2))
```
?
In this case, we may broadcast `OffsetArrays.Origin` over arrays to shift their origin. The syntax seems intuitive, and this behavior of `Origin` is not documented, so it'll not be a breaking change either.

**Comment by johnnychen94** [comment]
Looks good to me, and how about defining `Origin(A::AbstractArray)` as `Origin(first.(axes(A)))`?

**Comment by jishnub** [comment]
That makes sense. That way, we may transfer the origin from one array to another as `Origin(A)(B)`.

**Comment by johnnychen94** [comment]
closed in favor of #276

**Pull request closed without merging**